drivers/sensor/sht3xd/sht3xd.c

```diff
@@ -206,8 +211,31 @@ static int sht3xd_init(struct device *dev)
 	return 0;
 }
 
-struct sht3xd_data sht3xd_driver;
+#ifdef DT_SENSIRION_SHT3XD_0
 
-DEVICE_AND_API_INIT(sht3xd, DT_SHT3XD_NAME, sht3xd_init, &sht3xd_driver,
-		    NULL, POST_KERNEL, CONFIG_SENSOR_INIT_PRIORITY,
+struct sht3xd_data sht3xd0_driver;
+static const struct sht3xd_config sht3xd0_cfg = {
+	.bus_name = DT_SENSIRION_SHT3XD_0_BUS_NAME,
+#ifdef CONFIG_SHT3XD_TRIGGER
+	.alert_gpio_name = DT_SENSIRION_SHT3XD_0_ALERT_GPIOS_CONTROLLER,
+#endif
+	.base_address = DT_SENSIRION_SHT3XD_0_BASE_ADDRESS,
+#ifdef CONFIG_SHT3XD_TRIGGER
+	.alert_pin = DT_SENSIRION_SHT3XD_0_ALERT_GPIOS_PIN,
+#endif
+};
+
+DEVICE_AND_API_INIT(sht3xd0, DT_SENSIRION_SHT3XD_0_LABEL,
+		    sht3xd_init, &sht3xd0_driver, &sht3xd0_cfg,
+		    POST_KERNEL, CONFIG_SENSOR_INIT_PRIORITY,
 		    &sht3xd_driver_api);
+
+#else /* DT_SENSIRION_SHT3XD_0 */
+
+/* Hack to avoid defined-not-used errors in build_all testing */
```

Comment:
Why not define DT_SENSIRION_SHT3XD_0 in the build_all fixup instead? I realize we're trying to get rid of fixups, but I think build_all is a special case and I don't like this hack. I've wondered if we could maybe handle that special case differently by creating a dts overlay just for that test.

Reply:
Using a dts overlay for build_all would be nice, if it could be made to work.  Among other things it'd serve as examples for what's needed in the node.

Old workaround restored.